lib/MongoDB/Protocol/_Section.pm

```diff
@@ -0,0 +1,202 @@
+package MongoDB::Protocol::_Section;
```

Comment:
I think modeling this generically as an object is too expensive for the wire protocol pieces.